sample/src/main/java/com/thoughtworks/ark/sample/permission/PermissionSampleUI.kt

```diff
@@ -0,0 +1,254 @@
+import androidx.activity.compose.BackHandler
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.ColorFilter
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.window.Dialog
+import com.thoughtworks.ark.core.permission.MultiplePermissions
+import com.thoughtworks.ark.core.permission.PermissionHandler
+import com.thoughtworks.ark.core.permission.SinglePermission
+import com.thoughtworks.ark.sample.R
+import com.thoughtworks.ark.ui.component.AppFilledButton
+import com.thoughtworks.ark.ui.theme.Dimensions
+import com.thoughtworks.ark.ui.theme.LocalShapes
+import com.thoughtworks.ark.ui.theme.Theme
+
+@Composable
+fun PermissionSampleUI() {
+    val screenState = remember { mutableStateOf<ScreenState>(ScreenState.None) }
+    Box(
+        modifier = Modifier
+            .fillMaxSize()
+            .background(color = Theme.colors.background)
+            .padding(Dimensions.standardPadding)
+    ) {
+        when (screenState.value) {
+            is ScreenState.SinglePermissionState -> {
+                SinglePermissionScreen()
+            }
+            is ScreenState.SinglePermissionWithDialogState -> {
+                SinglePermissionWithDialogScreen()
+            }
+            is ScreenState.MultiplePermissionsState -> {
+                MultiplePermissionsScreen()
+            }
+            else -> {
+                MenuScreen(screenState = screenState)
+            }
+        }
+    }
+
+    BackHandler(screenState.value != ScreenState.None) {
+        if (screenState.value != ScreenState.None) {
+            screenState.value = ScreenState.None
+        }
+    }
+}
+
+sealed class ScreenState {
+    object None : ScreenState()
+    object SinglePermissionState : ScreenState()
+    object SinglePermissionWithDialogState : ScreenState()
+    object MultiplePermissionsState : ScreenState()
+}
+
+@Composable
```

Comment:
Could we extract the MenuScreen, SinglePermissionScreen, and State to some single files?

Reply:
fixed